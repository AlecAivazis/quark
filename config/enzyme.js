// external imports
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import browserEnv from 'browser-env'

// add a request animation frame polyfill
import raf from 'raf'
raf.polyfill()

// enable the react 16 adapter for enzyme
Enzyme.configure({ adapter: new Adapter() })

// add various browser mocks
browserEnv()
