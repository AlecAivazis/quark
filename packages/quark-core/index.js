export * from './components';
import * as _styles from './styles';
export { _styles as styles };