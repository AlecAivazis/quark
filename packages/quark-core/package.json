{
    "name": "quark-core",
    "version": "1.0.6",
    "module": "index.js",
    "main": "index.js",
    "dependencies": {
        "react-native-web": "0.3.4",
        "svgs": "^3.1.1"
    },
    "peerDependencies": {
        "react": "^16.3.0"
    }
}
