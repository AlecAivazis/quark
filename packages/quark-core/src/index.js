// @flow
export * from './components'
export * as styles from './styles'
