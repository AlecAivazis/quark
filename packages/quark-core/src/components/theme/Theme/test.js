// external imports
import React from 'react'
import { Text } from 'react-native'
import { mount } from 'enzyme'
import PropTypes from 'prop-types'
// local imports
import { Theme } from 'quark-core'

it('renders child', () =>
    expect(
        mount(
            <Theme>
                <Text>hello</Text>
            </Theme>
        )
    ).toMatchSnapshot())

it('provides a value for every color in a theme', () => {
    // the list of colors to look for
    const colors = [
        'primaryColor',
        'primaryColorLight',
        'primaryColorDark',
        'secondaryColor',
        'white',
        'grey1',
        'grey2',
        'grey3',
        'grey4',
        'grey5',
        'black',
        'red',
        'lightRed',
        'blue',
        'lightBlue',
        'green',
        'lightGreen'
    ]

    // the child to extract theme from context
    const Child = (args: {}, { theme }) => {
        // make sure we have
        return <Text>{JSON.stringify(theme)}</Text>
    }
    // pull the theme out of the context
    Child.contextTypes = {
        theme: PropTypes.object
    }

    // get the theme object
    const theme = JSON.parse(
        mount(
            <Theme>
                <Child />
            </Theme>
        ).text()
    )

    // map each color to their theme value and make sure we got the same number back
    expect(colors.map(color => theme[color]).filter(c => c)).toHaveLength(colors.length)
})

it('allows the theme to be customized', () => {
    // the child to extract theme from context
    const Child = (args: {}, { theme }) => {
        // make sure we have
        return <Text>{theme.primaryColor}</Text>
    }
    // pull the theme out of the context
    Child.contextTypes = {
        theme: PropTypes.object
    }

    // get the theme object
    expect(
        mount(
            <Theme theme={{ primaryColor: 'black' }}>
                <Child />
            </Theme>
        ).text()
    ).toEqual('black')
})
