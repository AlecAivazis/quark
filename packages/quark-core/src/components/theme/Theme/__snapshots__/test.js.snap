// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders child 1`] = `
<ThemeProvider>
  <Text>
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      hello
    </div>
  </Text>
</ThemeProvider>
`;
